[FIX] port base_gs1_barcode to v7

clonedagain opened the pull request:
Make base_gs1_barcode installable.
Relicence as AGPL-3.
Match coding conventions more closely
Use a distinct exception. This exception is placed in the package's scope to make it easier to use from other modules. Now use "from base_gs1_barcode import InvalidGS1Barcode"

Commit:
[FIX] port base_gs1_barcode to v7

Make base_gs1_barcode installable.
Use a distinct exception. This exception is placed in the package's scope to make it easier to use from other modules. Now use "from base_gs1_barcode import InvalidGS1Barcode"
Relicence as AGPL-3.
Match coding conventions more closely.
Fix typos.
Add translation template.

pedrobaeza commented on __unported__/base_gs1_barcode/__init__.py:
```diff
@@ -1,22 +1,25 @@
 ##############################################################################
 #
-#    This module is copyright (C) 2012 Numérigraphe SARL. All Rights Reserved.
+#    This module is copyright (C) 2014 Numérigraphe SARL. All Rights Reserved.
```
You should put 2012-2014

pedrobaeza replied:
All Rights Reserved is not valid anymore

pedrobaeza commented on __unported__/base_gs1_barcode/__init__.py:
```diff
@@ -1,22 +1,25 @@
 from . import res_users
+
+# Bring the main exception into the package's scope for easier reuse
+from .exceptions import InvalidGS1Barcode
```
Why do you need this here?

clonedagain replied:
So I can simply import it in other modules using "from adons.base_gs1_barcode import InvalidGS1Barcode"

pedrobaeza replied:
Thanks for the explanation.

pedrobaeza commented on __unported__/base_gs1_barcode/__openerp__.py:
```diff
@@ -1,19 +1,19 @@
 ##############################################################################
 #
-#    This module is copyright (C) 2012 Numérigraphe SARL. All Rights Reserved.
+#    This module is copyright (C) 2014 Numérigraphe SARL. All Rights Reserved.
```
Same remarks

pedrobaeza commented on __unported__/base_gs1_barcode/gs1_barcode.py:
```diff
@@ -1,19 +1,19 @@
 ##############################################################################
 #
-#    This module is copyright (C) 2012 Numérigraphe SARL. All Rights Reserved.
+#    This module is copyright (C) 2014 Numérigraphe SARL. All Rights Reserved.
```
Same remarks

pedrobaeza commented on __unported__/base_gs1_barcode/gs1_barcode.py:
```diff
@@ -24,44 +24,42 @@
-                                    ('numeric', 'Numeric value'),
-                                    ('date', 'Date') ],
-                                    'Data Type', required=1),
+        'ai': fields.char(
```
Label is missing (Application Identifer)

clonedagain commented:
@pedrobaeza all should be fixed.

pedrobaeza commented on base_gs1_barcode/res_users.py:
```diff
@@ -1,44 +1,42 @@
                  "or GS1-Datamatrix codes are scanned. No prefix is expected "
-                 "if this fields is left empty"
-            ),
+                 "if this fields is left empty"),
```
s/fields/field

pedrobaeza commented on base_gs1_barcode/res_users.py:
```diff
@@ -1,44 +1,42 @@
 
+
 # XXX This would probably be best in res_config_users
 class res_users(orm.Model):
```
ResUsers

pedrobaeza commented on base_gs1_barcode/res_users.py:
```diff
@@ -1,44 +1,42 @@
     _columns = {
-        # XXX those should be properties, not standard fields
         'gs1_barcode_prefix': fields.char(
             'Prefix', size=64,
```
Why is this at user level and not at company level?

clonedagain replied:
Because this needs to be configured the same as the on on the barcode reader.

pedrobaeza commented:
Please update also POT file.

clonedagain commented:
By the way I have the feeling that part of this could best served as python library, but I'm clueless about how to make it properly. If anyone has pointers to share I'll be glad to follow them (at least when I'll migrate to v8).

pedrobaeza commented:
Yeah, it could be, or make it as a kind of extensible framework for adding any kind of barcode, like we did on bank import modules from bank-statement-reconcile repo.

clonedagain commented:
Translation template added.

pedrobaeza commented:
Some minor issues but :+1:

clonedagain commented:
@pedrobaeza I didn't notice your later comments, I think it's fixed now.

pedrobaeza commented:
Thanks

yvaucher commented on base_gs1_barcode/i18n/fr.po:
```diff
@@ -0,0 +1,829 @@
+#. module: base_gs1_barcode
+#: model:gs1_barcode,name:base_gs1_barcode.AREA
+msgid "Product Area, in square meters"
+msgstr "Product Area, in square meters"
```
Please keep empty the translations when not translated

yvaucher commented:
I guess this has been in prod for some times @clonedagain ?

If you can empty the wrong french translantions I will proceed with merging

gurneyalex commented:
@clonedagain could you rebase the PR to trigger a new build?

Commit:
Merge remote-tracking branch 'upstream/7.0' into 7.0-enable-gs1

Commit:
[FIX] remove bogus translations

clonedagain commented:
@yvaucher @gurneyalex it's done, let's see what Travis thinks of it.

yvaucher commented:
@clonedagain Thanks, travis was fast, I think it only checked for PEP8 and lint.

Could it be that TESTS="1" is missing in travis config?

gurneyalex commented:
@clonedagain https://github.com/numerigraphe/stock-logistics-barcode/pull/1 should make runbot green

gurneyalex commented:
travis issue should be fixed by #13

Commit:
[FIX] no type on views

As Alexandre Fayolle noticed: "runbot is read because (...) 'Setting the type field is deprecated in the ir.ui.view model.'"

clonedagain commented:
I think it's fixed

yvaucher commented:
@clonedagain Was it intended to add all that text in commit message  7063a04 ?

Otherwise travis is green :+1:

clonedagain commented:
Yes I thought it illustrated the problem well, but I can cut it if it's 
a problem, just tell me.

clonedagain commented:
There was another view with a `type`, this new commit should fix it.

yvaucher commented:
@clonedagain Thanks

StefanRijnhart commented:
@clonedagain I was not aware of this PR, and I migrated the module to 8.0. See https://github.com/OCA/stock-logistics-barcode/pull/15. There are two details that I found were not as per specifications. Can you confirm those?

At the same time, I switched the custom exception to the conventional ValidationError, now available in openerp.exceptions.

I also modified the prefix and the separator to global configuration, but I understand now that that was a mistake. Will change that back.

StefanRijnhart commented:
Nice module, BTW!

StefanRijnhart commented:
:+1:, code review.
